src/integration-test/java/io/aiven/elasticsearch/gcs/GcsStoragePluginIT.java

```diff
@@ -0,0 +1,193 @@
+                        Files.newInputStream(gcsCredentialsPath)
+                );
+
+        final var storage = StorageOptions.newBuilder()
+                .setCredentials(credentials)
+                .build()
+                .getService();
+
+        final var batch = storage.batch();
+        storage.get(bucketName)
+                .list(Storage.BlobListOption.prefix(BASE_PATH + "/"))
+                .iterateAll()
+                .forEach(blob -> batch.delete(blob.getBlobId()));
+        batch.submit();
+    }
+
+
+    @Test
+    @Order(1)
+    void registerRepository() throws Exception {
+        final var node = clusterRunner.getNode(0);
+
+        final var enableRepositoryRequest =
+                "{ \"type\": \""
+                        + GcsBlobStoreRepository.TYPE + "\", "
+                        + "\"settings\": { \"bucket_name\": \""
+                        + bucketName + "\", \"base_path\": \"test_backup\" } "
+                        + "}";
+        try (final var response = EcrCurl.put(node, "/_snapshot/backup")
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.JSON_UTF_8.toString())
+                .body(enableRepositoryRequest)
+                .execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+        }
+
+        try (final var response = EcrCurl.get(node, "_snapshot/backup").execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+            final Map<String, Object> content = response.getContent(EcrCurl.jsonParser());
+            assertEquals(
+                    "{backup={settings={bucket_name=ples-test, base_path=" + BASE_PATH + "}, type=gcs-encrypted}}",
+                    content.toString()
+            );
+        }
+
+    }
+
+    @Test
+    @Order(2)
+    void createSnapshot() throws Exception {
+        final var node = clusterRunner.getNode(0);
+
+        for (int i = 0; i < 10_000; i++) {
+            try (final var curlResponse = EcrCurl.post(node, "/" + INDEX + "/_doc/")
+                    .header("Content-Type", "application/json")
+                    .body("{\"id\":\"200" + i + "\",\"msg\":\"test 200" + i + "\"}")
+                    .execute()) {
+                final Map<String, Object> content = curlResponse.getContent(EcrCurl.jsonParser());
+                assertNotNull(content);
+                assertEquals("created", content.get("result"));
+            }
+        }
+
+        try (final var response =
+                     EcrCurl.put(node, "/_snapshot/backup/snapshot_1?wait_for_completion=true")
+                             .execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+        }
+    }
+
+    @Test
+    @Order(3)
+    void restoreSnapshot() throws Exception {
+        final var node = clusterRunner.getNode(0);
+        try (final var response = EcrCurl.get(node, "/_snapshot/backup/snapshot_1").execute()) {
+            assertEquals(200, response.getHttpStatusCode());
+        }
+    }
```

Comment:
This doesn't test restoration.

It would be nice if we check that the end-to-end backup taking and restoration works like: index deletion, restoration and checking documents in the restored index.

Reply:
good one ... will add